### PR TITLE
Feature pokemon detail

### PR DESCRIPTION
**Pull request by Abhishek-Raj-AT**
Summary: 
Add function in Pokemon Details and MyPokemonList
Change log:
Added function in Pokemon Details and MyPokemonList
 test cases
- [ ] Check the API integration in the network tab
- [ ] Check whether ViewDetails links are working or not
- [ ] Check catch button is working or not
- [ ] Clicked on the catch button then we should get a successful popup message
- [ ] Check when the user writes in the input box and clicked the catch button then shows the data in the MyPokemon list
- [ ] InPokemonList remove button is working or not
- [ ] When clicking on the remove button the item should remove

### CODE REVIEW

File: src/config/api.tsx [hunk 1]
```diff
@@ -5,11 +5,7 @@ const api ={
         battle: "ability/battle-armor",
         pokemon: "pokemon",
         pidgeot: "pokemon/{image_id}/",
-    },
-    imageEndPoint: {
-        backDefault: "back/1.png",
-        bacjShiny: "shiny/1.png",
-        frontDefault: "other/home/shiny/1.png",
+        ability: "ability/ability_id/",
```
Comment (bhavikg-dev) [inline src/config/api.tsx:8]: This is not the right approach to set id

Refer the management code how it is handled

File: src/Service/PokemonListService.tsx [hunk 2]
```diff
@@ -23,3 +24,13 @@ export async function getImageDetails(payload: GetImageList) {
         return hasError(error)
     }
 }
+export async function getAbilityDetails(payload: GetAbilityList) {
+    const endPoint = api.endPoint.ability.replace("ability_id", payload.id?.toString());
```
Comment (bhavikg-dev) [inline src/Service/PokemonListService.tsx:28]: This is not the right approach to set id

Refer the management code how it is handled

File: src/Redux/PokemonListSlice/index.tsx [hunk 3]
```diff
@@ -58,26 +65,29 @@ const pokemonSlice = createSlice({
...
+              order
             };
           } else {
             state.imagePokemonList = initialImage;
           }
           state.isLoading = false;
         }
       )
-      .addCase(getImageDetailsAction.rejected, (state: PokemonList) => {
+      .addCase(getPOkemonDetailsAction.rejected, (state: PokemonList) => {
```
Comment (bhavikg-dev) [inline src/Redux/PokemonListSlice/index.tsx:90]: `getPOkemonDetailsAction`
Is this correct ?

File: src/Redux/PokemonDetailSlice/DetailsType.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+export interface Version{
+    name: string;
+    url: string;
+}
+
+export interface DetailsList {
+  list: Details[];
+  isLoading: boolean;
+  name: string;
+  is_main_series: boolean;
```
Comment (bhavikg-dev) [inline src/Redux/PokemonDetailSlice/DetailsType.tsx:40]: What is the use of this ?

**Review by bhavikg-dev — CHANGES_REQUESTED** [review]

File: src/Redux/PokemonListSlice/index.tsx (hunk 3 above)
Comment (Abhishek-Raj-AT) [reply src/Redux/PokemonListSlice/index.tsx:90]: No sir, I changed the name

File: src/Service/PokemonListService.tsx (hunk 2 above)
Comment (Abhishek-Raj-AT) [reply src/Service/PokemonListService.tsx:28]: I solved this issue as I take a reference the management portal code

File: src/Redux/PokemonDetailSlice/DetailsType.tsx (hunk 4 above)
Comment (Abhishek-Raj-AT) [reply src/Redux/PokemonDetailSlice/DetailsType.tsx:40]: I forgot to remove. so I removed this

**Review by bhavikg-dev — APPROVED** [review]

**Pull request merged**